tinyagentos/routes/project_doc_review.py

```
@@ -0,0 +1,112 @@
+    if isinstance(auth, JSONResponse):
+        return auth
+    actor_id, _is_agent, _project = auth
+
+    try:
+        review = await store.set_review_state(
+            project_id, doc_path, payload.state, actor_id
+        )
+    except ValueError as exc:
+        detail = str(exc)
+        if "invalid transition" in detail:
+            raise HTTPException(status_code=409, detail=detail)
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)